src/controllers/discussionController.ts

```diff
@@ -0,0 +1,266 @@
+import { Request, Response, NextFunction } from "express";
+import Discussion, { IDiscussion } from "../models/Discussion";
+import { StatusCodes } from "http-status-codes";
+import {
+  BadRequestError,
+  NotFoundError,
+  UnauthenticatedError,
+} from "../errors";
+import Joi from "joi";
+import mongoose from "mongoose";
+import { IUser } from "../models/User";
+
+const discussionJoiSchema = Joi.object({
```

Comment:
you may have a validation file, 
either a validation.ts for all validations or a discussionValidation.ts for just this.

Reply:
fixed